doc/design/s3-undelete.md

```diff
@@ -0,0 +1,69 @@
+# **S3 Undelete**
```

Comment:
```suggestion
# **S3 Trailing Delete**
```

Per my comment on slack, I think that "S3 Undelete" is a bit of mis-leading title for this development since it is not really about "undelete" feature, which is already similarly possible in current S3 versioned setup without GC, but rather about establishing policy for delayed pruning/GC for keys on the level of S3.
Since "trailing delete" was used below (although I found no good hits googling for it) -- I am suggesting it here.  But may be even it could be something like

```suggestion
# **S3 Delayed GC Setup**
```

which would
- better describe what is presented in the document
- accent on relation to GC

Reply:
I agree that "trailing delete" is a better description for this. While the primary motivator for this feature is doing garbage collection, I think the title of "S3 Delayed GC Setup" is too low-level, especially considering that I don't provide a precise definition for "Garbage Collection" anywhere in this doc. I'll go ahead and change it to "S3 Trailing Delete", but let me know if you have any further thoughts on this.